### PR TITLE
Refactoring BaseButton with respect to how it processes menuProps

### PR DESCRIPTION
**Pull request by whykushal93**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes
As I was working on a different change that is related to menus in BaseButton, I noticed that it was confusing to reason about the `menuProps` property in the BaseButton's state. 

1. Initially, we used it as a proxy mainly to determine whether the menu was hidden or not. Other reasons for its existence were for determining `shouldFocusOnContainer` and `shouldFocusOnMount` properties. 
2. However, with persisted menus it resulted in some complications. Now, we needed to not only see if `menuProps` was defined, we also needed to see if it `hidden` was set to true/false.
3. The code is not consistent as to which `menuProps` should override the other. In some cases we have used the props, in at least one case we have used the state (in `doNotLayer` case)
4. Also, in a lot of places in the code, we need to first check if `menuProps` in the state is defined. 

To fix all of the above, 
1.  I removed `menuProps` from the state. Instead added a `menuHidden` to the state.
2.  `shouldFocusOnMount` and `shouldFocusOnContainer` are private fields on the BaseButton class, which are only used when rendering a menu.
3. Because of 2, there is one place in the code where I needed to **use `forceUpdate`**. I think the code in that part looks cleaner than before in my opinion.  If this is unacceptable, I can add `shouldFocusOnMount` and `shouldFocusOnContainer` to the state.

#### Focus areas to test
1. I tested normal menus and persisted of menus works as before.. 
2. I tested for normal menus focus works correctly.
3. The code where I had to use `forceUpdate` was put in place because of [this](https://github.com/OfficeDev/office-ui-fabric-react/issues/8351) issue in PR [8555](https://github.com/OfficeDev/office-ui-fabric-react/pull/8555). I tried testing that scenario and it seems like there is no change in behavior that I see. But, the current behavior itself seems buggy. @khmakoto could you take a look at current behavior in Fabric and tell me if its correct ?

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10524)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ContextualMenu/__snapshots__/ContextualMenu.test.tsx.snap [hunk 1]
```diff
@@ -4,7 +4,7 @@ exports[`ContextualMenu ContextualMenu snapshot ContextualMenu should be present
 <button
   aria-expanded={false}
   aria-haspopup={true}
-  aria-owns="id__94-menu"
+  aria-owns={null}
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/__snapshots__/ContextualMenu.test.tsx.snap]: Is this change expected? I don't see any changes to the test, so this looks like a regression.

Also, this seems to be exposing a pre-existing issue in button. Unused ARIA attributes should be `undefined`, not `null`.

Comment (whykushal93) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/__snapshots__/ContextualMenu.test.tsx.snap]: It is not completely ok for this to change. But, I think it should be fine, This snapshot is taken on a menu which is persisted but hidden. In this case, the menu is not shown to the user, but present in the DOM with visibility:hidden. 

In that case, I think it should not matter if aria-owns is defined or underfined, afaik.

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 2]
```diff
@@ -599,10 +603,11 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
   }
 
   private _onRenderSplitButtonMenuButton(classNames: ISplitButtonClassNames | undefined, keytipAttributes: any): JSX.Element {
-    const { allowDisabledFocus, checked, disabled, splitButtonMenuProps } = this.props;
+    const { allowDisabledFocus, checked, disabled, splitButtonMenuProps, splitButtonAriaLabel } = this.props;
+    const { menuHidden } = this.state;
     let menuIconProps = this.props.menuIconProps;
 
-    const { splitButtonAriaLabel } = this.props;
+    const {} = this.props;
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Nit: remove this line.

**Comment by micahgodbolt** [comment]
i think @dzearing  was involved in the original decision to base menu visibility off of menu props. I think your approach is great, but I'd want to check to see he recalls why we went with our existing approach.

**Comment by whykushal93** [comment]
Here is the issue for the CommandButton with menu and shouldFocusOnMount:false - https://github.com/OfficeDev/office-ui-fabric-react/issues/10654

**Comment by whykushal93** [comment]
@natalieethell I feel like I am inclined to merge this than wait longer. I have been blocked on making more changes for the feature I have been working because of this PR

**Pull request merged**